### PR TITLE
Add verifier "type of x == type of output" to `Poly_MulConstantOp`

### PR DESCRIPTION
**Pull request by mr0re1**

### CODE REVIEW

File: include/Dialect/Poly/IR/PolyOps.td [hunk 1]
```diff
@@ -47,6 +47,8 @@ def Poly_MulConstantOp : Poly_Op<"mul_constant", [ElementwiseMappable]> {
   let results = (outs
     PolynomialLike:$output
   );
+
+  let hasVerifier = 1;
```
Comment (j2kun) [inline include/Dialect/Poly/IR/PolyOps.td:51]: Consider instead using the trait `AllTypesMatch<["x", "output"]>` which avoids the need for any custom C++

**Review by j2kun — APPROVED** [review]
One minor suggestion otherwise LGTM

File: include/Dialect/Poly/IR/PolyOps.td (hunk 1 above)
Comment (mr0re1) [reply include/Dialect/Poly/IR/PolyOps.td:51]: Thank you! I was almost sure that `AllTypesMatch` exists )

Comment (mr0re1) [reply include/Dialect/Poly/IR/PolyOps.td:51]: I will update assembly format as well.

**Review by j2kun — APPROVED** [review]

**Pull request merged**